python/ray/worker.py

```diff
@@ -668,6 +669,8 @@ def init(
         driver_mode = LOCAL_MODE
     else:
         driver_mode = SCRIPT_MODE
+        if job_config:
+            ray_pkg.update_runtime_env(job_config)
```

Comment:
Trying to think of how to make this more clear, since we have this two-stage upload process for the driver. How about something like:

```suggestion
            # Rewrite the URI. Note the package isn't uploaded to the URI until later in the init
            # process, after we assign the job id to the driver.
            ray_pkg.rewrite_working_dir_uri(job_config)
```

Ideally the driver work isn't split across two distinct functions, but not sure if that's easy to fix.

Reply:
If we can connect to GCS and upload it without `connect`, we can put them in the same place. The ordering now is 1) calculate pkg URI; 2) init core worker with this updated runtime; 3) upload to GCS.

If we can do: 1) -> 3) -> 2) it'll be great.

I'll update it according to your suggestion. Please make it as resolved if you think for now it's OK.